### PR TITLE
feat: add enter app lock passcode UI and logic [WPB-4695]

### PR DESCRIPTION
**Pull request by saleniuk**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4695" title="WPB-4695" target="_blank">StoryWPB-4695</a>  [Android] Toggle app lock from the app settings (Password only)
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Screen that allows user to enter the app lock passcode.
Logic to keep track of when the app is being put into background and to show such screen after specific time passed.

#### How to Test

In privacy settings set the app lock passcode and put the app into background.

### Attachments (Optional)

In the video below the timestamp was changed from 60s to 5s to make it easier to record and show this logic.

https://github.com/wireapp/wire-android-reloaded/assets/30429749/73d2ad4d-d6c1-49f7-ad6d-73de65c7555b

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1033](https://10.10.124.134/job/AR-build-pipeline/1033/) **failed**.

**Comment by AndroidBob** [comment]
Build [1034](https://10.10.124.134/job/AR-build-pipeline/1034/) **failed**.

**Comment by AndroidBob** [comment]
Build [1036](https://10.10.124.134/job/AR-build-pipeline/1036/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7686532)(7686532)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7686532)(7686532)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/appLock/LockCodeTimeManager.kt [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                observeAppLockConfigUseCase()
+            ) { appVisibilityTimestampData, appLockConfig ->
+                appVisibilityTimestampData.isAppVisible
+                        && appLockConfig !is AppLockConfig.Disabled
+                        && appVisibilityTimestampData.timestamp >= 0
+                        && (System.currentTimeMillis() - appVisibilityTimestampData.timestamp) > (appLockConfig.timeoutInSeconds * 1000)
+            }
+                .distinctUntilChanged()
+        }
+        .shareIn(scope = appCoroutineScope, started = SharingStarted.Eagerly, replay = 1)
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/ui/home/appLock/LockCodeTimeManager.kt]: Would be nice to test this guy

**Review by vitorhugods — APPROVED** [review]
Looks good.

Just wonder if LockCodeTimeManager will have some tests.

File: app/src/main/kotlin/com/wire/android/ui/home/appLock/LockCodeTimeManager.kt (hunk 1 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/home/appLock/LockCodeTimeManager.kt]: Tests added 😃

**Comment by AndroidBob** [comment]
Build [1227](https://10.10.124.134/job/AR-build-pipeline/1227/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7804573)(7804573)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7804573)(7804573)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [1228](https://10.10.124.134/job/AR-build-pipeline/1228/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7804964)(7804964)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7804964)(7804964)-staging-compat.apk)

**Review by alexandreferris — APPROVED** [review]
loooking 👌🏻

**Review by gongracr — APPROVED** [review]
🔒 👏 LGTM 🍪

**Pull request merged**